packages/dev/core/src/Lights/volumetricSpotLight.ts

```diff
@@ -0,0 +1,80 @@
+import { SpotLight } from "./spotLight";
+import { Scene } from "core/scene";
+import { Axis } from "core/Maths/math.axis";
+import { CreateCylinder } from "core/Meshes/Builders/cylinderBuilder";
+import { Mesh } from "core/Meshes/mesh";
+import { ShaderMaterial } from "core/Materials/shaderMaterial";
+import "../Shaders/volumetricSpot.fragment";
+import "../Shaders/volumetricSpot.vertex";
+
+export interface IVolumetricSpotLight {
+    diameterTop: number;
+    diameterBottom: number;
+    rayLength?: number;
+    scene: Scene;
+}
+
+export class VolumetricSpotLight implements IVolumetricSpotLight {
+    diameterTop: number;
+    diameterBottom: number;
+    rayLength?: number;
+    scene: Scene;
+    public spotLight: SpotLight;
+    public volumetricMaterial: ShaderMaterial;
+    public spotLightCone: Mesh;
+
+    constructor(spotLight: SpotLight, diameterTop: number, diameterBottom: number, rayLength: number, scene: Scene) {
+        this.spotLight = spotLight;
+        this.diameterTop = diameterTop;
+        this.diameterBottom = diameterBottom;
+        this.rayLength = rayLength;
+        this.scene = scene;
+        this.createVolumetricSpotLight();
+
+        this.scene.registerBeforeRender(() => {

```

Comment:
You should save the returned observer (and use `onBeforeRenderObservable` instead of `registerBeforeRender`) and remove it in a `dispose` method.